### PR TITLE
contiki: Add documentation on public functions of contiki

### PR DESCRIPTION
**Pull request by zehortigoza**
Signed-off-by: José Roberto de Souza <jose.souza@intel.com>

### CODE REVIEW

File: src/lib/common/include/sol-mainloop-contiki.h [hunk 1]
```diff
@@ -39,7 +39,26 @@
 extern "C" {
 #endif
 
+/**
+ * Set the event and data that woke up the Soletta protothread,
+ * so this event can be internally handled.
+ * Should only be used on Contiki implementation of SOL_MAIN_DEFAULT().
```
Comment (mbelluzzo) [inline src/lib/common/include/sol-mainloop-contiki.h]: Plz add a @warning on this

File: src/lib/common/include/sol-mainloop-contiki.h [hunk 2]
```diff
@@ -39,7 +39,26 @@
 extern "C" {
 #endif
 
+/**
+ * Set the event and data that woke up the Soletta protothread,
```
Comment (mbelluzzo) [inline src/lib/common/include/sol-mainloop-contiki.h]: @brief is needed here. Leave a blank line after handled. so the "should" isn't part of @brief.

File: src/lib/common/include/sol-mainloop-contiki.h [hunk 3]
```diff
@@ -39,7 +39,26 @@
...
+ *
+ * @param data the data of the event
+ *
+ * @see SOL_MAIN_DEFAULT()
+ */
 void sol_mainloop_contiki_event_set(process_event_t ev, process_data_t data);
+
+/**
+ * Do a single mainloop iteration.
+ * Should only be used on Contiki implementation of SOL_MAIN_DEFAULT().
```
Comment (mbelluzzo) [inline src/lib/common/include/sol-mainloop-contiki.h]: The same for this one.

**Comment by mbelluzzo** [comment]
+1, imho you can apply with the proposed changes without a v2. Thanks!

**Comment by zehortigoza** [comment]
Merged with the requested changes.

**Pull request closed without merging**